config/routes.rb

```diff
@@ -36,4 +36,12 @@
   get "/veterinary_offices/:veterinary_office_id/veterinarians", to: "veterinary_offices#veterinarians"
   get "/veterinary_offices/:veterinary_office_id/veterinarians/new", to: "veterinarians#new"
   post "/veterinary_offices/:veterinary_office_id/veterinarians", to: "veterinarians#create"
+
+  get "/pet_applications/new" , to: "pet_applications#new"
+  post "/pet_applications", to: "pet_applications#create"
+  get "/pet_applications/:id", to: "pet_applications#show"
+  patch "/pet_applications/:id", to: "pet_applications#update"
+
+  post "/pet_applications/:app_id/pets/:pet_id", to: "pet_application_pets#create"
```

Comment:
I think I might have been the one to give you this last route. This route should actually be `"/pet_applications/:app_id/pets"` and you would get the ID in the params hash either from a hidden field or something similar.

All the other routes are restful.